bundles/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,254 @@
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.pixometer.internal.config.ReadingInstance;
+import org.openhab.binding.pixometer.internal.serializer.CustomReadingInstanceDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link MeterHandler} is responsible for handling data and measurements of a meter thing
+ *
+ * @author Jerome Luckenbach - Initial contribution
+ */
+public class MeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeterHandler.class);
+
+    private static final String API_VERSION = "v1";
+    private static final String API_METER_ENDPOINT = "meters";
+    private static final String API_READINGS_ENDPOINT = "readings";
+
+    private final GsonBuilder gsonBuilder = new GsonBuilder().registerTypeAdapter(ReadingInstance.class,
+            new CustomReadingInstanceDeserializer());
+    private final Gson gson = gsonBuilder.create();
+
+    private String resourceID;
+    private String meterID;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public MeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No handling needed, since we have only values to read.
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Pixometer handler '{}'", getThing().getUID());
+        updateStatus(ThingStatus.UNKNOWN);
+
+        try {
+            Configuration config = getThing().getConfiguration();
+            setRessourceID((String) config.get(CONFIG_THING_RESSOURCE_ID));
+
+            Bridge b = this.getBridge();
+            if (b == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "Could not find bridge (pixometer config). Did you choose one?");
+                return;
+            }
+
+            String token = new StringBuilder("Bearer ")
+                    .append(b.getConfiguration().get(CONFIG_BRIDGE_AUTH_TOKEN).toString()).toString();
```

Comment:
Hey @Hilbrand,

I am refactoring the config parts to a `getConfigurationAs` based solution as you suggested.
Could you help me out here maybe?

I have not yet figured out how to get the config for a bridge parsed in a class here.
From what i read `getConfigAs` is bound to the current thing, which is in my case a meter thing.
Or am i overseeing something?!

Reply:
Yes the config is bound to the bridge and that is also a hint to what might be changed here. The bridge is the proxy between pixometer and the meters. So the bridge should make the api call and pass on the results to the meter handlers. The handlers should not make the call themselves. In this case it means each meter would call independently, whil all data is returned. So 3 meters would result in 3 calls, while it should be only 1.

Reply:
Thanks for the explanation.
Is this a veto critera or would it be ok to postpone it for now?

I am of course interested in improving the methods, but would also like to get the first version (and this PR) done now.
This would cause some major adaptions again, which i would like to do afterwards.